clients/cobol-lsp-vscode-extension/src/commands/RunAnalysisCLI.ts

```diff
@@ -0,0 +1,251 @@
+   */
+  protected buildJavaCommand(currentFileLocation: string) {
+    const extensionFolder: string | undefined =
+      this.extensionUri.fsPath + "/server/jar/server.jar";
+
+    if (extensionFolder && currentFileLocation !== "") {
+      return (
+        'java -jar "' +
+        extensionFolder +
+        '" ' +
+        this.buildAnalysisCommandPortion(currentFileLocation)
+      );
+    }
+
+    return "";
+  }
+
+  /**
+   * Provides the portion of the command from "analysis" onwards.
+   * Is the same for both the Java and Native builds.
+   * @param currentFileLocation - Location of the main cobol file being analyzed.
+   * @protected
+   */
+  protected buildAnalysisCommandPortion(currentFileLocation: string) {
+    const copyBookCommand = `-cf=${
+      this.copybookConfigLocation === ""
+        ? "."
+        : '"' + this.copybookConfigLocation + '"'
+    }`;
+
+    return 'analysis -s "' + currentFileLocation + '" ' + copyBookCommand;
+  }
+
+  /**
+   * Sends a given command to a terminal.
+   * Checks to see if one named "Analysis" is already created, if so clear and reuse it.
+   * @param command - The command to run from the terminal.
+   * @protected
+   */
+  protected sendToTerminal(command: string) {
+    const existingTerminal = vscode.window.terminals.find(
+      (term: Terminal) => term.name === "Analysis",
+    );
+    const terminal = existingTerminal
+      ? existingTerminal
+      : vscode.window.createTerminal("Analysis");
+
+    terminal.sendText(command);
+    terminal.show(true);
+  }
+
+  /**
+   * Retrieves the location of the currently open file.
+   * If it is not saved, creates a temporary one on the disk to send to the command line with a copy of the contents.
+   *    - Does not result in the actual file being saved to the end user.
+   * @protected
+   */
+  protected async getCurrentFileLocation() {
+    if (vscode.window.activeTextEditor) {
+      if (vscode.window.activeTextEditor.document.uri.scheme === "untitled") {
+        return this.saveTempFile();
+      } else {
+        return vscode.window.activeTextEditor.document.uri.fsPath;
+      }
+    }
+    return "";
+  }
+
+  /**
+   * Creates a temporary one on the disk to send to the command line with a copy of the contents.
+   *    - Does not result in the actual file being saved to the end user.
+   * @protected
+   */
+  protected async saveTempFile() {
+    const data = vscode.window.activeTextEditor?.document.getText();
+
+    if (data) {
+      const fileStatResult = await vscode.workspace.fs.stat(
+        this.globalStorageUri,
+      );
+      if (fileStatResult.type === vscode.FileType.Directory) {
+        const tempFolderUri = vscode.Uri.joinPath(
+          this.globalStorageUri,
+          "/tempAnalysisFiles",
+        );
+        try {
+          const tempFolderStatResult = await vscode.workspace.fs.stat(
+            tempFolderUri,
+          );
+          if (tempFolderStatResult.type === vscode.FileType.Directory) {
```

Comment:
According to the docs `vscode.workspace.fs.createDirectory` behaves as `mkdir -p`, so it should be enough to just
```javascript
const tempFolderUri = vscode.Uri.joinPath(this.globalStorageUri, "tempAnalysisFiles");
await vscode.workspace.fs.createDirectory(tempFolderUri);
```
without the additional checks and then just proceed with the deletions and writeFile.

Reply:
Implemented and tested in the latest commit.